command/results_utils.go

```diff
@@ -1,61 +1,139 @@
+	// }
+
+	logger.Task().Info("Attach test results succeeded")
+
+	return nil
+}
+
+// sendTestLogsAndResults sends the test logs and test results to the API
+// server and Cedar.
+func sendTestLogsAndResults(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *model.TaskConfig, logs []model.TestLog, results [][]task.TestResult) error {
 	td := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
+	// ship all of the test logs off to the server
+	logger.Task().Info("Sending test logs to server...")
 
-	for i, res := range results.Results {
+	allResults := task.LocalTestResults{}
+	for idx, log := range logs {
 		if ctx.Err() != nil {
-			return errors.Errorf("operation canceled during uploading")
+			return errors.New("operation canceled")
 		}
 
-		if res.LogRaw != "" {
-			logger.Execution().Info("Attaching raw test logs")
-			testLogs := &model.TestLog{
-				Name:          res.TestFile,
-				Task:          conf.Task.Id,
-				TaskExecution: conf.Task.Execution,
-				Lines:         []string{res.LogRaw},
-			}
-
-			id, err := comm.SendTestLog(ctx, td, testLogs)
-			if err != nil {
-				logger.Execution().Errorf("problem posting raw logs from results %s", err.Error())
-			} else {
-				results.Results[i].LogId = id
-			}
-
-			// clear the logs from the TestResult struct after it has been saved in the test logs. Since they are
-			// being saved in the test_logs collection, we can clear them to prevent them from being saved in the task
-			// collection.
-			results.Results[i].LogRaw = ""
+		logId, err := comm.SendTestLog(ctx, td, &log)
+		if err != nil {
+			// continue on error to let the other logs be posted
+			logger.Task().Errorf("problem posting log: %v", err)
+		}
+
+		// add all of the test results that correspond to that log to the
+		// full list of results
+		for _, result := range results[idx] {
+			result.LogId = logId
+
+			allResults.Results = append(allResults.Results, result)
 		}
 	}
-	logger.Execution().Info("attaching test results")
+	logger.Task().Info("Finished posting logs to server")
+
+	logger.Task().Info("Sending parsed results to server...")
+
+	if err := comm.SendTestResults(ctx, td, &allResults); err != nil {
+		logger.Task().Errorf("problem posting parsed results to the server: %+v", err)
+		return errors.Wrap(err, "problem sending test results")
+	}
+
+	// TODO (EVG-7780): send test results for projects that enable it.
+	// if err := sendTestResultsToCedar(ctx, conf.Task, comm, &allResults); err != nil {
+	//     return errors.Wrap(err, "sending test results to Cedar")
+	// }
+
+	logger.Task().Info("Successfully sent parsed results to server")
 
-	err := comm.SendTestResults(ctx, client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}, results)
+	return nil
+}
+
+// sendTestResultsToCedar sends the given test results to Cedar.
+func sendTestResultsToCedar(ctx context.Context, t *task.Task, comm client.Communicator, results *task.LocalTestResults) error {
```

Comment:
what about logging to cedar?

Reply:
What do you mean by this?

Reply:
test logs should also go to cedar

Reply:
It seems like there's no support in the cedar for sending test logs right now, so that would seemingly be future work.